packages/arb-avm-cpp/avm/src/machinestate/machinestate.cpp

```diff
@@ -39,9 +39,11 @@ MachineState::MachineState()
 
 MachineState::MachineState(const std::vector<CodePoint>& code_,
                            const value& static_val_,
-                           std::shared_ptr<TuplePool>& pool_)
-    : pendingInbox(pool_.get()), context({0, 0}), inbox(pool_.get()) {
-    pool = pool_;
+                           std::shared_ptr<TuplePool> pool_)
+    : pool(std::move(pool_)),
+      pendingInbox(pool_.get()),
+      context({0, 0}),
+      inbox(pool_.get()) {
```

Comment:
`pool_` has been moved into `pool` so it is now invalid. Use `pool.get()` instead